### PR TITLE
Do not create files when optional collection is empty

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

File: src/ntfs/write.rs [hunk 1]
```diff
@@ -285,6 +294,22 @@ where
 }
 
 pub fn write_codes(path: &path::Path, collections: &Collections) -> Result<()> {
+    fn collection_has_codes<T: Codes>(collection: &CollectionWithId<T>) -> bool {
```
Comment (TeXitoi) [inline src/ntfs/write.rs]: bad name, it should be `collection_has_no_codes`, or negate the function.

File: src/ntfs/write.rs [hunk 2]
```diff
@@ -326,6 +351,18 @@ where
 }
 
 pub fn write_object_properties(path: &path::Path, collections: &Collections) -> Result<()> {
+    fn collection_has_object_properties<T: Properties>(collection: &CollectionWithId<T>) -> bool {
```
Comment (TeXitoi) [inline src/ntfs/write.rs]: Same here

**Comment by TeXitoi** [comment]
You now need to rebase :stuck_out_tongue:

**Comment by datanel** [comment]
arf

**Pull request merged**